tests/commands/test_rename.py

```diff
@@ -124,6 +134,9 @@ def test_duplicate_new_names(tmp_library: TemporaryLibrary,
         # First document will rename, but all others should raise a warning
         assert len(caplog.records) == doc_count - 1
         assert all(
-            os.path.basename(record.args[0]) == "same-name"
+            # FIXME: Why do we know that args will be a tuple whose first
+            # component is a string? Is there a better way of obtaining this
+            # information?
+            os.path.basename(record.args[0]) == "same-name"       # type: ignore
```

Comment:
The record here is just a `LogRecord` which has documented arguments:
https://docs.python.org/3/library/logging.html#logrecord-attributes
You might just need to check that `args` is non-empty instead?

Reply:
Hm, and they're all guaranteed to be castable to `str`. I'll see if asserting `length(record.args)>0` and that `record.args[0] is not None` works.

Reply:
... Nope -- `record.args` has type `tuple[object, ...] | Mapping[str, object] | None`

Reply:
so it's not even guaranteed to have positional arguments

Reply:
OK, just shoved type assertions in there to keep `mypy` happy.